src/main/java/jarvis/logic/parser/AddConsultCommandParser.java

```diff
@@ -0,0 +1,58 @@
+package jarvis.logic.parser;
+
+import static jarvis.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static jarvis.logic.parser.CliSyntax.PREFIX_END_DATE_TIME;
+import static jarvis.logic.parser.CliSyntax.PREFIX_LESSON;
+import static jarvis.logic.parser.CliSyntax.PREFIX_START_DATE_TIME;
+import static jarvis.logic.parser.CliSyntax.PREFIX_STUDENT_INDEX;
+
+import java.time.LocalDateTime;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import jarvis.commons.core.index.Index;
+import jarvis.logic.commands.AddConsultCommand;
+import jarvis.logic.parser.exceptions.ParseException;
+import jarvis.model.LessonDesc;
+import jarvis.model.TimePeriod;
+
+/**
+ * Parses input arguments and creates a new AddConsultCommand object
+ */
+public class AddConsultCommandParser {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddConsultCommand
+     * and returns an AddConsultCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddConsultCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_LESSON, PREFIX_START_DATE_TIME,
+                PREFIX_END_DATE_TIME, PREFIX_STUDENT_INDEX);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_LESSON, PREFIX_START_DATE_TIME, PREFIX_END_DATE_TIME,
+                PREFIX_STUDENT_INDEX) || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddConsultCommand.MESSAGE_USAGE));
+        }
+
+        LessonDesc consultDesc = ParserUtil.parseLessonDesc(argMultimap.getValue(PREFIX_LESSON).get());
+        LocalDateTime startDateTime = ParserUtil.parseDateTime(argMultimap.getValue(PREFIX_START_DATE_TIME).get());
+        LocalDateTime endDateTime = ParserUtil.parseDateTime(argMultimap.getValue(PREFIX_END_DATE_TIME).get());
+        TimePeriod consultPeriod = new TimePeriod(startDateTime, endDateTime);
```

Comment:
Need to check that startDateTime is before endDateTime and throw an exception if this is not the case. Can use [`LocalDateTime::isBefore`](https://docs.oracle.com/javase/8/docs/api/java/time/LocalDateTime.html#isBefore-java.time.chrono.ChronoLocalDateTime-).

Same for the other lesson commands.